lib/config/index.js

```diff
@@ -150,66 +145,56 @@ for (let i = keys.length; i--;) {
   }
 }
```

Comment:
we could start getting rid of this as well. Or maybe convict has a facility of dealing with this? would it throw an error if something unexpected was configured? if not, we should look into if it _could_ throw errors or at least warnings.